health/target.go

```diff
@@ -0,0 +1,105 @@
+package health
+
+import (
+	"sync"
+
+	t "github.com/zenoss/zenoss-go-sdk/health/target"
+)
+
+func newRawHealth(target *t.Target, metricIDs []string) *rawHealth {
+	tHealth := &rawHealth{
+		target:        target,
+		healthy:       true,
+		heartBeat:     false,
+		counters:      make(map[string]int32),
+		totalCounters: make(map[string]int32),
+		rawMetrics:    make(map[string][]float64),
+		messages:      make([]*t.Message, 0),
+	}
+	for _, metricID := range metricIDs {
+		tHealth.rawMetrics[metricID] = make([]float64, 0)
+	}
+	return tHealth
+}
+
+// rawHealth is a struct that keeps target information together with raw target health
+type rawHealth struct {
+	target        *t.Target
+	healthy       bool
```

Comment:
Should we allow for healthy, degrade, unhealthy to cover more scenarios where the component is impaired, but not out?

Reply:
Fixed: HealthStatus type defines healthy, degrade, unhealthy component states